prebid-server/developers/add-a-module-go.md

```diff
@@ -0,0 +1,266 @@
+
+The supported stages are described in the [general module overview](/prebid-server/developers/add-a-module.html#2-understand-the-endpoints-and-stages) and in PBS-Core source code at the "github.com/prebid/prebid-server/hooks" package.
+
+These are the available hooks that can be implemented in a module:
+
+- github.com/prebid/prebid-server/hooks/hookstage.Entrypoint
+- github.com/prebid/prebid-server/hooks/hookstage.RawAuctionRequest
+- github.com/prebid/prebid-server/hooks/hookstage.ProcessedAuctionRequest
+- github.com/prebid/prebid-server/hooks/hookstage.BidderRequest
+- github.com/prebid/prebid-server/hooks/hookstage.RawBidderResponse
+- github.com/prebid/prebid-server/hooks/hookstage.AllProcessedBidResponses
+- github.com/prebid/prebid-server/hooks/hookstage.AuctionResponse
+
+In a module it is not necessary to implement all mentioned interfaces but at least one is required by your functionality.
+
+### Examples
+
+1) To **update** the request in the `BidderRequest` hook you would return:
+```
+import (
+	"context"
+
+	"github.com/prebid/prebid-server/hooks/hookstage"
+)
+
+type Module struct{}
+
+func (m Module) HandleBidderRequestHook(
+	ctx context.Context,
+	invocationCtx hookstage.ModuleInvocationContext,
+	payload hookstage.BidderRequestPayload,
+) (hookstage.HookResult[hookstage.BidderRequestPayload], error) {
+    changeSet := hookstage.ChangeSet[hookstage.BidderRequestPayload]{}
+    changeSet.BidderRequest().BAdv().Update([]string{"a.com"})
+	
+    return hookstage.HookResult[hookstage.BidderRequestPayload]{ChangeSet: changeSet}, nil
+}
+```
+
+Please note, the `hookstage.ChangeSet` has a restricted set of methods, but methods can be easily extended when more use cases come up.
+
+For more complex payload updates, you can choose another method:
+```
+func (m Module) HandleBidderRequestHook(
+	ctx context.Context,
+	invocationCtx hookstage.ModuleInvocationContext,
+	payload hookstage.BidderRequestPayload,
+) (hookstage.HookResult[hookstage.BidderRequestPayload], error) {
+    battrByImp := map[string][]adcom1.CreativeAttribute{"imp_ID1": []adcom1.CreativeAttribute{adcom1.AttrAudioAuto}}
+    changeSet := hookstage.ChangeSet[hookstage.BidderRequestPayload]{}
+    changeSet.AddMutation(func(payload hookstage.BidderRequestPayload) (hookstage.BidderRequestPayload, error) {
+        for i, imp := range payload.BidRequest.Imp {
+            if battr, ok := battrByImp[imp.ID]; ok {
+                imp.Banner.BAttr = battr
+                payload.BidRequest.Imp[i] = imp
+            }
+        }
+        return payload, nil
+    }, hookstage.MutationUpdate, "bidrequest", "imp", "banner", "battr")
+	
+    return hookstage.HookResult[hookstage.BidderRequestPayload]{ChangeSet: changeSet}, nil
+}
+```
+
+2) To **reject** the bidder in the `BidderRequest` hook you would return:
```

Comment:
How about:
"To **reject** the bidder in the `BidderRequest` hook, your hook implementation would return a hook result with a reject flag and an NBR code:"